File: api-certvalidation/src/tr/gov/tubitak/uekae/esya/api/certificate/validation/find/certificate/CertificateFinderFromLDAP.java
package tr.gov.tubitak.uekae.esya.api.certificate.validation.find.certificate;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import tr.gov.tubitak.uekae.esya.api.asn.x509.EAuthorityInfoAccessSyntax;
import tr.gov.tubitak.uekae.esya.api.asn.x509.ECertificate;
import tr.gov.tubitak.uekae.esya.api.certificate.validation.BaglantiUtil;
import tr.gov.tubitak.uekae.esya.api.common.ESYAException;

import java.util.List;

/**
 * Finds issuer certificate according to Authority Info Access (AIA) extension
 * information. It only searches in LDAP addresses 
 * @author IH
 */
public class CertificateFinderFromLDAP extends CertificateFinderFromAIA
{
	private static Logger logger = LoggerFactory.getLogger(CertificateFinderFromLDAP.class);

    private static final String DIZIN_TIPI = "dizintipi";

    /**
     * Sertifikanin LDAP uzerindeki SM Sertifikasi dagitici adreslerini doner
     */
    protected List<String> _getAddresses(ECertificate aSertifika)
    {
        EAuthorityInfoAccessSyntax aia = aSertifika.getExtensions().getAuthorityInfoAccessSyntax();
        if (aia != null) {
            return aia.getLdapAddresses();
        }
        return null;
    }

    /**
     * verilen LDAP adresinden Sertifika bilgisini okur
     */
    protected ECertificate _getCertificate(String aAdres)
    {
        String dizinTipi = mParameters.getParameterAsString(DIZIN_TIPI);
        byte [] certData =  BaglantiUtil.dizindenVeriOku(aAdres, dizinTipi);
        if(certData != null)
        {
        	ECertificate cert = null;
			try 
			{
				cert = new ECertificate(certData);
			} 
			catch (ESYAException e) 
			{
				logger.debug("Certificate Asn1 decode error", e);
			}
        	return cert;
        }
        else
        {
        	logger.debug("LDAP'tan veri okunamadı.");
        }
        return null;
    }

}

File: api-certvalidation/csharp/src/find/certificate/CertificateFinderFromLDAP.cs
using System;
using System.Collections.Generic;
using System.Reflection;
using log4net;
using tr.gov.tubitak.uekae.esya.api.asn.x509;
using tr.gov.tubitak.uekae.esya.api.common;

namespace tr.gov.tubitak.uekae.esya.api.certificate.validation.find.certificate
{
    /**
     * Finds issuer certificate according to Authority Info Access (AIA) extension
     * information. It only searches in LDAP addresses 
     * @author IH
     */
    public class CertificateFinderFromLDAP : CertificateFinderFromAIA
    {
        private static readonly ILog logger = LogManager.GetLogger(MethodBase.GetCurrentMethod().DeclaringType);
        private static readonly String DIZIN_TIPI = "dizintipi";

        /**
         * Sertifikanın LDAP üzerindeki SM Sertifikası dağıtıcı adreslerini döner
         */
        protected override List<String> _getAddresses(ECertificate aSertifika)
        {
            EAuthorityInfoAccessSyntax aia = aSertifika.getExtensions().getAuthorityInfoAccessSyntax();
            if (aia != null)
            {
                return aia.getLdapAddresses();
            }
            return null;
        }

        /**
         * verilen LDAP adresinden Sertifika bilgisini okur
         */
        //override protected byte[] _getCertificateData(String aAdres)
        //{
        //    //String dizinTipi = mParameters.getParameterAsString(DIZIN_TIPI);
        //    return BaglantiUtil.dizindenVeriOku(aAdres/*, dizinTipi*/);
        //}
        
        /**
         * verilen LDAP adresinden Sertifika bilgisini okur
         */
        protected override ECertificate _getCertificate(String aAdres)
        {
            try
            {
                String dizinTipi = mParameters.getParameterAsString(DIZIN_TIPI);
                byte[] certData = BaglantiUtil.dizindenVeriOku(aAdres, dizinTipi);
                ECertificate cert = new ECertificate(certData);
                return cert;
            }
            catch (Exception e)
            {
                logger.Debug(String.Format("{0} LDAP adresinden sertifika alınamadı.", aAdres), e);
            }
            
            return null;
        }

    }

}
